lms/djangoapps/courseware/models.py

```diff
@@ -248,3 +251,56 @@ class Meta(object):   # pylint: disable=missing-docstring
 
     field = models.CharField(max_length=255)
     value = models.TextField(default='null')
+
+
+SCORE_CHANGED = Signal(
+    providing_args=[
+        'points_possible',  # Maximum score available for the exercise
+        'points_earned',   # Score obtained by the user
+        'user',  # User object representing the student
+        'course',  # CourseLocator object representing the course
+        'usage'  # BlockUsageLocator indicating the courseware instance
+    ]
+)
+
+
+@receiver(score_set)
+def submissions_score_set_handler(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Consume the score_set signal defined in the Submissions API, and convert it
+    to a SCORE_CHANGED signal defined in this module. Converts the unicode keys
+    for user, course and item into the standard representation for the
+    SCORE_CHANGED signal.
+    """
+    user = user_by_anonymous_id(kwargs['anonymous_user_id'])
```

Comment:
If you _must_ use `**kwargs`, you must use a safe access here. That means doing

`kwargs.get('anonymous_user_id', None)`

(the second argument to `.get` is the default value you want to provide in case the dictionary doesn't have the provided key -- see https://docs.python.org/2/library/stdtypes.html#dict.get )